src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3587,121 +3589,97 @@ internal void VerifyValueType(object? value)
             }
 
             Debug.Fail("Error result not expected");
-            return false;
+            return default;
         }
 
         private CheckValueStatus TryChangeType(
             ref object? value,
             out bool copyBack,
             out bool isValueType)
         {
-            // If this is a ByRef get the element type and check if it's compatible
-            if (IsByRef)
+            RuntimeType? sigElementType;
+            if (RuntimeTypeHandle.TryGetByRefElementType(this, out sigElementType))
             {
-                RuntimeType sigElementType = RuntimeTypeHandle.GetElementType(this);
-
-                // If a nullable, pass the object even though it's a value type.
-                if (sigElementType.IsNullableOfT)
-                {
-                    // Treat as a reference type since a null value may be replaced with T or vise-versa.
-                    isValueType = false;
-                    copyBack = true;
-                    return CheckValueStatus.Success;
-                }
+                copyBack = true;
+                Debug.Assert(!sigElementType.IsGenericParameter);
 
                 if (sigElementType.IsInstanceOfType(value))
                 {
-                    // Need to create an instance of the ByRef if null was provided, but only if primitive, enum or value type
-                    value = AllocateValueType(sigElementType, value, fForceTypeChange: false);
-                    isValueType = sigElementType.IsValueType;
-                    copyBack = true;
-                    return CheckValueStatus.Success;
-                }
-
-                if (value == null)
-                {
-                    if (IsByRefLike)
+                    Debug.Assert(!sigElementType.IsGenericParameter);
+                    isValueType = RuntimeTypeHandle.IsValueType(sigElementType);
+                    if (sigElementType.IsValueType)
```

Comment:
Is this going to be different from the check immediately above it? `isValueType = RuntimeTypeHandle.IsValueType(sigElementType);`?

Reply:
No, thanks. Look like an incomplete refactor. Calling `RuntimeTypeHandle.IsValueType` is faster.